workflow/controller/workflowpod.go

```diff
@@ -636,11 +650,11 @@ func (woc *wfOperationCtx) createVolumes(tmpl *wfv1.Template) []apiv1.Volume {
 			},
 		})
 	}
+
+	volumes = append(volumes, volumeVarArgo)
+
 	switch woc.getContainerRuntimeExecutor() {
-	case common.ContainerRuntimeExecutorKubelet, common.ContainerRuntimeExecutorK8sAPI, common.ContainerRuntimeExecutorPNS:
-	case common.ContainerRuntimeExecutorEmissary:
-		volumes = append(volumes, volumeVarArgo)
-	default:
+	case "", common.ContainerRuntimeExecutorDocker:
```

Comment:
I added it like this because of the same switch/case in [`workflowpod.go Line 538`](https://github.com/argoproj/argo-workflows/pull/6714/files#diff-0cd2c830c5575d814bc0dccc445c0483f0ea0411b72862784714ed44c6806280L538). 

Due to the change of default executor this is wrong. Current master would be wrong in this case too, though. 

I changed now both instances and adjusted tests accordingly. If you prefer I could also make a separate PR for those two changes.